src/wallet/rpc/spend.cpp

```diff
@@ -715,6 +716,12 @@ CreatedTransactionResult FundTransaction(CWallet& wallet, const CMutableTransact
         coinControl.m_max_tx_weight = options["max_tx_weight"].getInt<int>();
     }
 
+    if (tx.version == TRUC_VERSION) {
+        if (!coinControl.m_max_tx_weight.has_value() || coinControl.m_max_tx_weight.value() > 40000) {
```

Comment:
Instead of magic numbers, try using `TRUC_MAX_VSIZE * WITNESS_SCALE_FACTOR`

Reply:
Also, there is a separate size limit for TRUC children - I don't think that's been implemented yet?

Reply:
Yes, I will look into implementing the separate size limit.